pkg/analyzer/nodes/node.go

```diff
@@ -0,0 +1,211 @@
+package nodes
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	clair "github.com/stackrox/scanner"
+	"github.com/stackrox/scanner/database"
+	"github.com/stackrox/scanner/pkg/analyzer"
+	"github.com/stackrox/scanner/pkg/component"
+	"github.com/stackrox/scanner/pkg/matcher"
+	"github.com/stackrox/scanner/pkg/metrics"
+	"github.com/stackrox/scanner/singletons/requiredfilenames"
+)
+
+// Information about extracted files.
+type fileMetadata struct {
+	// If true, the file has executable permissions.
+	isExecutable bool
+	// If true, contents can be extracted from the filesystem.
+	isExtractable bool
+}
+
+// filesMap is an analyzer.Files implementation mapping files extracted from
+// filesystem directories. In this implementation the file content is read lazily
+// when calls to Get() are made. If an error occurs during content read the map
+// will stop returning entries and the error will be available at readErr().
+// Analyzer are expected to handle missing files gracefully during analysis and
+// check readErr() to confirm all potential data was read.
+type filesMap struct {
```

Comment:
can we add a line

```
var _ analyzer.Files = (*filesMap)(nil)
```

See https://github.com/uber-go/guide/blob/master/style.md#verify-interface-compliance

Reply:
+1